### PR TITLE
DM-38870: Change how file datastore transfer_from handles absolute URIs

### PR DESCRIPTION
**Pull request by timj**
Previously the code stored the absolute URI in the target datastore. Now it only does that in "auto" transfer mode. In other modes it will copy/link etc.

## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/core/storedFileInfo.py [hunk 1]
```diff
@@ -244,3 +241,12 @@ def from_record(cls: Type[StoredFileInfo], record: Mapping[str, Any]) -> StoredF
             dataset_id=record["dataset_id"],
         )
         return info
+
+    def update(self, **kwargs: Any) -> StoredFileInfo:
```
Comment (andy-slac) [inline python/lsst/daf/butler/core/storedFileInfo.py:245]: Unrecognized keywords are just silently ignored?

File: python/lsst/daf/butler/registry/bridge/monolithic.py [hunk 2]
```diff
@@ -237,7 +237,13 @@ def join_records(
         # Run query, transform results into a list of dicts that we can later
         # use to delete.
         with self._db.query(info_in_trash) as sql_result:
-            rows = [dict(**row, datastore_name=self.datastoreName) for row in sql_result.mappings()]
+            # can not use dict(**row, datastore_name=) because mypy knows that
```
Comment (andy-slac) [inline python/lsst/daf/butler/registry/bridge/monolithic.py]: Can you do `dict(row, datastore=...)` instead?

**Review by andy-slac — APPROVED** [review]
Looks good, couple of minor questions.

File: python/lsst/daf/butler/registry/bridge/monolithic.py (hunk 2 above)
Comment (timj) [reply python/lsst/daf/butler/registry/bridge/monolithic.py]: Very good. That does work.

**Pull request merged**